internal/planner/mapper/targetable.go

```diff
@@ -81,16 +72,15 @@ type ObjectProperty struct {
 	Name string
 }
 
-func (k *ObjectProperty) GetProp(data any) any {
+func (k *ObjectProperty) PropertyAndOperator(data any, defaultOp string) (any, string, error) {
 	if data == nil {
-		return nil
+		return nil, defaultOp, nil
 	}
-	object := data.(map[string]any)
-	return object[k.Name]
-}
-
-func (k *ObjectProperty) GetOperatorOrDefault(defaultOp string) string {
-	return defaultOp
+	_, ok := data.(core.Doc)
```

Comment:
suggestion:  Made me smile a little to see you do this again in this PR - and I'm curious as to why/where you think I prefer this :grin: (are you thinking back to me preferring nil checks vs failed casting?)

My suggestion is the same as the last, code is simpler IMO, and avoids having to care about an extra, somewhat irrelevant type:
```go
docMap, ok := data.(map[string]any)
if !ok {
  return nil, defaultOp, NewErrFieldOrAliasNotFound(k.Name)
}
return docMap[k.Name], defaultOp, nil
```

Reply:
I think I burned that one into my brain. Should be fixed now.